test/e2e/node/pods.go

```diff
@@ -415,6 +415,160 @@ var _ = SIGDescribe("Pods Extended", func() {
 		})
 	})
 
+	ginkgo.Describe("Pod Generation", func() {
+		var podClient *e2epod.PodClient
+		ginkgo.BeforeEach(func() {
+			podClient = e2epod.NewPodClient(f)
+		})
+
+		ginkgo.It("pod generation should start at 1 and increment per update", func(ctx context.Context) {
+			ginkgo.By("creating the pod")
+			name := "pod-generation-" + string(uuid.NewUUID())
+			value := strconv.Itoa(time.Now().Nanosecond())
+			pod := e2epod.NewAgnhostPod(f.Namespace.Name, name, nil, nil, nil)
+			pod.ObjectMeta.Labels = map[string]string{
+				"time": value,
+			}
+			pod.Spec.InitContainers = []v1.Container{{
+				Name:    "init-container",
+				Image:   "busybox:1.28",
+				Command: []string{"sh", "-c", "exit 0"},
+			}}
+
+			ginkgo.By("setting up selector")
+			selector := labels.SelectorFromSet(labels.Set(map[string]string{"time": value}))
+			options := metav1.ListOptions{LabelSelector: selector.String()}
+			pods, err := podClient.List(ctx, options)
+			framework.ExpectNoError(err, "failed to query for pod")
+			gomega.Expect(pods.Items).To(gomega.BeEmpty())
```

Comment:
I think you can remove this, seems irrelevant.

Reply:
done